api/src/main/java/org/xnio/XnioWorker.java

```diff
@@ -1284,6 +1289,42 @@ public Thread run() {
         }
     }
 
+    class CleanableWorkerThreadFactory implements ThreadFactory {
```

Comment:
This does not need to be a separate thread factory.  This behavior should always be enabled.